{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Using Connectivity map data and PharmacoGx R package to find drugs against coronaviruses"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Connectivity map or Cmap is resource that uses gene expression data to connect disease transcriptomic signature with those of different types of cells after drug treatment. If the transcriptomic signature of a drug on cell line is opposite from disease signature, it can be expected that this drug could have a positive effect on amelioration of disease.\n",
    "\n",
    "\n",
    "References:\n",
    "Subramanian A, et al. A Next Generation Connectivity Map: L1000 Platform And The First 1,000,000 Profiles. Cell. 2017/12/1. 171(6):1437–1452.\n",
    "\n",
    "Lamb J, et al. The Connectivity Map: using gene-expression signatures to connect small molecules, genes, and disease. Science. 2006/9/29. 313(5795):1929-35, (2006)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Harmonizome is a collection of processed datasets from over 70 major online resources. I downloaded from Harmonizome \"GEO Signatures of Differentially Expressed Genes for Viral Infections\". For this analysis I used transcriptomics data of lungs from mice that were infected with icSARS CoV (GEO accesion number GSE50000;7th day of infection). For this analysis I havent used transcriptomic data of novel human coronavirus 2019 or SARS-2, but according to the literature all of these viruses (MERS,SARS1 and SARS2) are betacoronaviruses that have origin in bats and SARS-2 is shown to be very similar to SARS-01 according to phylogenic analysis. \n",
    "\n",
    "References:\n",
    "Rouillard AD, Gundersen GW, Fernandez NF, Wang Z, Monteiro CD, McDermott MG, Ma'ayan A. The harmonizome: a collection of processed datasets gathered to serve and mine knowledge about genes and proteins. Database (Oxford). 2016 Jul 3;2016. pii: baw100.\n",
    "\t\n",
    "Josset L, Menachery VD, Gralinski LE, Agnihothram S et al. Cell host response to infection with novel human coronavirus EMC predicts potential antivirals and important differences with SARS coronavirus. mBio 2013 Apr 30;4(3):e00165-13. PMID: 23631916\n",
    "\n",
    "Menachery VD, Eisfeld AJ, Schäfer A, Josset L et al. Pathogenic influenza viruses and coronaviruses utilize similar and contrasting approaches to control interferon-stimulated gene responses. mBio 2014 May 20;5(3):e01174-14. PMID: 24846384\n",
    "\n",
    "Yushun Wan, Jian Shang, Rachel Graham, Ralph S. Baric, Fang Li. Receptor recognition by novel coronavirus from Wuhan: An analysis based on decade-long structural studies of SARS. Journal of Virology, 2020; DOI: 10.1128/JVI.00127-20\n",
    "\n",
    "Yadi Zhou, Yuan Hou, Jiayu Shen, Yin Huang, William Martin, Feixiong Cheng.Network-based Drug Repurposing for Human Coronavirus\n",
    "doi: https://doi.org/10.1101/2020.02.03.20020263"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "For the analysis of Connectivity map data I used PharmacoGx R package.\n",
    "\n",
    "P Smirnov, Z Safikhani, N El-Hachem , D Wang, A She , C Olsen, MFreeman, H Selby, D Gendoo , P Grossman , A Beck , H Aerts ,\n",
    "M Lupien, A Goldenberg ,B Haibe-Kains PharmacoGx: an R package for analysis of large pharmacogenomic datasets Bioinformatics (2016)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [],
   "source": [
    "library(PharmacoGx)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 11 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>PSet.Name</th><th scope=col>Dataset.Type</th><th scope=col>Available.Molecular.Profiles</th><th scope=col>Date.Updated</th><th scope=col>URL</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>CCLE_2013</th><td>CCLE_2013      </td><td>sensitivity </td><td>rna/mutation                </td><td>Tue Sep 15 18:50:07 2015</td><td>https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/CCLE_Nature2013.RData</td></tr>\n",
       "\t<tr><th scope=row>CCLE</th><td>CCLE           </td><td>sensitivity </td><td>rna/rnaseq/mutation/cnv     </td><td>Thu Dec 10 18:17:14 2015</td><td>https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/CCLE.RData           </td></tr>\n",
       "\t<tr><th scope=row>GDSC_2013</th><td>GDSC_2013      </td><td>sensitivity </td><td>rna/mutation                </td><td>Mon Oct  5 16:07:54 2015</td><td>https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/CGP_Nature2013.RData </td></tr>\n",
       "\t<tr><th scope=row>GDSC</th><td>GDSC           </td><td>sensitivity </td><td>rna/rna2/mutation/fusion/cnv</td><td>Wed Dec 30 10:44:21 2015</td><td>https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/GDSC.RData           </td></tr>\n",
       "\t<tr><th scope=row>GDSC1000</th><td>GDSC1000       </td><td>sensitivity </td><td>rna                         </td><td>Thu Aug 25 11:13:00 2016</td><td>https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/GDSC1000.RData       </td></tr>\n",
       "\t<tr><th scope=row>gCSI</th><td>gCSI           </td><td>sensitivity </td><td>rnaseq/cnv                  </td><td>Mon Jun 13 18:50:12 2016</td><td>https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/gCSI.RData           </td></tr>\n",
       "\t<tr><th scope=row>FIMM</th><td>FIMM           </td><td>sensitivity </td><td>                            </td><td>Mon Oct 3 17:14:00 2016 </td><td>https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/FIMM.RData           </td></tr>\n",
       "\t<tr><th scope=row>CTRPv2</th><td>CTRPv2         </td><td>sensitivity </td><td>                            </td><td>Thu Aug 25 11:15:00 2016</td><td>https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/CTRPv2.RData         </td></tr>\n",
       "\t<tr><th scope=row>CMAP</th><td>CMAP           </td><td>perturbation</td><td>rna                         </td><td>Mon Sep 21 02:38:45 2015</td><td>https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/CMAP.RData           </td></tr>\n",
       "\t<tr><th scope=row>L1000_compounds</th><td>L1000_compounds</td><td>perturbation</td><td>rna                         </td><td>Mon Jan 25 12:51:00 2016</td><td>https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/L1000_compounds.RData</td></tr>\n",
       "\t<tr><th scope=row>L1000_genetic</th><td>L1000_genetic  </td><td>perturbation</td><td>rna                         </td><td>Mon Jan 25 12:51:00 2016</td><td>https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/L1000_genetic.RData  </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 11 × 5\n",
       "\\begin{tabular}{r|lllll}\n",
       "  & PSet.Name & Dataset.Type & Available.Molecular.Profiles & Date.Updated & URL\\\\\n",
       "  & <fct> & <fct> & <fct> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\tCCLE\\_2013 & CCLE\\_2013       & sensitivity  & rna/mutation                 & Tue Sep 15 18:50:07 2015 & https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/CCLE\\_Nature2013.RData\\\\\n",
       "\tCCLE & CCLE            & sensitivity  & rna/rnaseq/mutation/cnv      & Thu Dec 10 18:17:14 2015 & https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/CCLE.RData           \\\\\n",
       "\tGDSC\\_2013 & GDSC\\_2013       & sensitivity  & rna/mutation                 & Mon Oct  5 16:07:54 2015 & https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/CGP\\_Nature2013.RData \\\\\n",
       "\tGDSC & GDSC            & sensitivity  & rna/rna2/mutation/fusion/cnv & Wed Dec 30 10:44:21 2015 & https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/GDSC.RData           \\\\\n",
       "\tGDSC1000 & GDSC1000        & sensitivity  & rna                          & Thu Aug 25 11:13:00 2016 & https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/GDSC1000.RData       \\\\\n",
       "\tgCSI & gCSI            & sensitivity  & rnaseq/cnv                   & Mon Jun 13 18:50:12 2016 & https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/gCSI.RData           \\\\\n",
       "\tFIMM & FIMM            & sensitivity  &                              & Mon Oct 3 17:14:00 2016  & https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/FIMM.RData           \\\\\n",
       "\tCTRPv2 & CTRPv2          & sensitivity  &                              & Thu Aug 25 11:15:00 2016 & https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/CTRPv2.RData         \\\\\n",
       "\tCMAP & CMAP            & perturbation & rna                          & Mon Sep 21 02:38:45 2015 & https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/CMAP.RData           \\\\\n",
       "\tL1000\\_compounds & L1000\\_compounds & perturbation & rna                          & Mon Jan 25 12:51:00 2016 & https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/L1000\\_compounds.RData\\\\\n",
       "\tL1000\\_genetic & L1000\\_genetic   & perturbation & rna                          & Mon Jan 25 12:51:00 2016 & https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/L1000\\_genetic.RData  \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 11 × 5\n",
       "\n",
       "| <!--/--> | PSet.Name &lt;fct&gt; | Dataset.Type &lt;fct&gt; | Available.Molecular.Profiles &lt;fct&gt; | Date.Updated &lt;fct&gt; | URL &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|\n",
       "| CCLE_2013 | CCLE_2013       | sensitivity  | rna/mutation                 | Tue Sep 15 18:50:07 2015 | https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/CCLE_Nature2013.RData |\n",
       "| CCLE | CCLE            | sensitivity  | rna/rnaseq/mutation/cnv      | Thu Dec 10 18:17:14 2015 | https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/CCLE.RData            |\n",
       "| GDSC_2013 | GDSC_2013       | sensitivity  | rna/mutation                 | Mon Oct  5 16:07:54 2015 | https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/CGP_Nature2013.RData  |\n",
       "| GDSC | GDSC            | sensitivity  | rna/rna2/mutation/fusion/cnv | Wed Dec 30 10:44:21 2015 | https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/GDSC.RData            |\n",
       "| GDSC1000 | GDSC1000        | sensitivity  | rna                          | Thu Aug 25 11:13:00 2016 | https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/GDSC1000.RData        |\n",
       "| gCSI | gCSI            | sensitivity  | rnaseq/cnv                   | Mon Jun 13 18:50:12 2016 | https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/gCSI.RData            |\n",
       "| FIMM | FIMM            | sensitivity  | <!----> | Mon Oct 3 17:14:00 2016  | https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/FIMM.RData            |\n",
       "| CTRPv2 | CTRPv2          | sensitivity  | <!----> | Thu Aug 25 11:15:00 2016 | https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/CTRPv2.RData          |\n",
       "| CMAP | CMAP            | perturbation | rna                          | Mon Sep 21 02:38:45 2015 | https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/CMAP.RData            |\n",
       "| L1000_compounds | L1000_compounds | perturbation | rna                          | Mon Jan 25 12:51:00 2016 | https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/L1000_compounds.RData |\n",
       "| L1000_genetic | L1000_genetic   | perturbation | rna                          | Mon Jan 25 12:51:00 2016 | https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/L1000_genetic.RData   |\n",
       "\n"
      ],
      "text/plain": [
       "                PSet.Name       Dataset.Type Available.Molecular.Profiles\n",
       "CCLE_2013       CCLE_2013       sensitivity  rna/mutation                \n",
       "CCLE            CCLE            sensitivity  rna/rnaseq/mutation/cnv     \n",
       "GDSC_2013       GDSC_2013       sensitivity  rna/mutation                \n",
       "GDSC            GDSC            sensitivity  rna/rna2/mutation/fusion/cnv\n",
       "GDSC1000        GDSC1000        sensitivity  rna                         \n",
       "gCSI            gCSI            sensitivity  rnaseq/cnv                  \n",
       "FIMM            FIMM            sensitivity                              \n",
       "CTRPv2          CTRPv2          sensitivity                              \n",
       "CMAP            CMAP            perturbation rna                         \n",
       "L1000_compounds L1000_compounds perturbation rna                         \n",
       "L1000_genetic   L1000_genetic   perturbation rna                         \n",
       "                Date.Updated            \n",
       "CCLE_2013       Tue Sep 15 18:50:07 2015\n",
       "CCLE            Thu Dec 10 18:17:14 2015\n",
       "GDSC_2013       Mon Oct  5 16:07:54 2015\n",
       "GDSC            Wed Dec 30 10:44:21 2015\n",
       "GDSC1000        Thu Aug 25 11:13:00 2016\n",
       "gCSI            Mon Jun 13 18:50:12 2016\n",
       "FIMM            Mon Oct 3 17:14:00 2016 \n",
       "CTRPv2          Thu Aug 25 11:15:00 2016\n",
       "CMAP            Mon Sep 21 02:38:45 2015\n",
       "L1000_compounds Mon Jan 25 12:51:00 2016\n",
       "L1000_genetic   Mon Jan 25 12:51:00 2016\n",
       "                URL                                                                                 \n",
       "CCLE_2013       https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/CCLE_Nature2013.RData\n",
       "CCLE            https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/CCLE.RData           \n",
       "GDSC_2013       https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/CGP_Nature2013.RData \n",
       "GDSC            https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/GDSC.RData           \n",
       "GDSC1000        https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/GDSC1000.RData       \n",
       "gCSI            https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/gCSI.RData           \n",
       "FIMM            https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/FIMM.RData           \n",
       "CTRPv2          https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/CTRPv2.RData         \n",
       "CMAP            https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/CMAP.RData           \n",
       "L1000_compounds https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/L1000_compounds.RData\n",
       "L1000_genetic   https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/L1000_genetic.RData  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "availablePSets()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [],
   "source": [
    "CMAP.sigs <- downloadPertSig(\"CMAP\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [],
   "source": [
    " drug.perturbation <- CMAP.sigs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [],
   "source": [
    "sars1_up_and_down_genes <-read.csv(\"icSARS_Cov_Day7_None_GSE50000.csv\",stringsAsFactors = F, header = T, sep=\";\")\n",
    "rownames(sars1_up_and_down_genes) <- sars1_up_and_down_genes$Feature\n",
    "sars1<- sars1_up_and_down_genes$Direction\n",
    "names(sars1) <- rownames(sars1_up_and_down_genes)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [],
   "source": [
    "res_sars1 <- apply(drug.perturbation[,,c(\"tstat\", \"fdr\")], \n",
    "             2,function(x, sars1){\n",
    "         return(connectivityScore(x=x, \n",
    "                                  y=sars1,\n",
    "                               method=\"fgsea\", nperm=100))\n",
    "       },sars1 = sars1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'Connectivity'</li><li>'P_value'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'Connectivity'\n",
       "\\item 'P\\_value'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'Connectivity'\n",
       "2. 'P_value'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"Connectivity\" \"P_value\"     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "rownames(res_sars1) <- c(\"Connectivity\", \"P_value\")\n",
    "res_sars1 <- t(res_sars1)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [],
   "source": [
    "con_score_sars <-res_sars1[order((res_sars1[,1]),decreasing = F),]\n",
    "p_value_sars <- res_sars1[order((res_sars1[,2]),decreasing = F),]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A matrix: 1288 × 2 of type dbl</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Connectivity</th><th scope=col>P_value</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>valproic acid</th><td>-0.376625</td><td>0.003243431</td></tr>\n",
       "\t<tr><th scope=row>rifabutin</th><td>-0.340380</td><td>0.003249652</td></tr>\n",
       "\t<tr><th scope=row>levomepromazine</th><td>-0.329905</td><td>0.003363063</td></tr>\n",
       "\t<tr><th scope=row>promazine</th><td>-0.338460</td><td>0.003408985</td></tr>\n",
       "\t<tr><th scope=row>Prestwick-559</th><td>-0.332285</td><td>0.003453743</td></tr>\n",
       "\t<tr><th scope=row>maprotiline</th><td>-0.345995</td><td>0.003483930</td></tr>\n",
       "\t<tr><th scope=row>nicergoline</th><td>-0.297975</td><td>0.003540995</td></tr>\n",
       "\t<tr><th scope=row>GW-8510</th><td>-0.365775</td><td>0.003643482</td></tr>\n",
       "\t<tr><th scope=row>lasalocid</th><td>-0.287965</td><td>0.003653011</td></tr>\n",
       "\t<tr><th scope=row>pyrvinium</th><td>-0.312920</td><td>0.003692544</td></tr>\n",
       "\t<tr><th scope=row>nortriptyline</th><td>-0.346990</td><td>0.003785772</td></tr>\n",
       "\t<tr><th scope=row>prochlorperazine</th><td>-0.360245</td><td>0.003809080</td></tr>\n",
       "\t<tr><th scope=row>norcyclobenzaprine</th><td>-0.319960</td><td>0.003935323</td></tr>\n",
       "\t<tr><th scope=row>fluphenazine</th><td>-0.351740</td><td>0.004089757</td></tr>\n",
       "\t<tr><th scope=row>LY-294002</th><td>-0.358135</td><td>0.004118938</td></tr>\n",
       "\t<tr><th scope=row>raloxifene</th><td>-0.301795</td><td>0.004218595</td></tr>\n",
       "\t<tr><th scope=row>4,5-dianilinophthalimide</th><td>-0.809330</td><td>0.006282675</td></tr>\n",
       "\t<tr><th scope=row>trichostatin A</th><td>-0.330245</td><td>0.006335983</td></tr>\n",
       "\t<tr><th scope=row>eucatropine</th><td>-0.288555</td><td>0.006421429</td></tr>\n",
       "\t<tr><th scope=row>syrosingopine</th><td>-0.318895</td><td>0.006459413</td></tr>\n",
       "\t<tr><th scope=row>mebendazole</th><td>-0.300325</td><td>0.006618701</td></tr>\n",
       "\t<tr><th scope=row>lycorine</th><td> 0.319895</td><td>0.006644780</td></tr>\n",
       "\t<tr><th scope=row>rescinnamine</th><td>-0.323295</td><td>0.006987249</td></tr>\n",
       "\t<tr><th scope=row>reserpine</th><td>-0.285080</td><td>0.007184183</td></tr>\n",
       "\t<tr><th scope=row>sirolimus</th><td>-0.361685</td><td>0.007393165</td></tr>\n",
       "\t<tr><th scope=row>trifluoperazine</th><td>-0.375195</td><td>0.007405285</td></tr>\n",
       "\t<tr><th scope=row>vorinostat</th><td>-0.378310</td><td>0.008683363</td></tr>\n",
       "\t<tr><th scope=row>methotrexate</th><td>-0.336405</td><td>0.008864197</td></tr>\n",
       "\t<tr><th scope=row>fulvestrant</th><td>-0.281245</td><td>0.009065718</td></tr>\n",
       "\t<tr><th scope=row>phenazopyridine</th><td>-0.321425</td><td>0.009136916</td></tr>\n",
       "\t<tr><th scope=row>...</th><td>...</td><td>...</td></tr>\n",
       "\t<tr><th scope=row>PF-00562151-00</th><td>0</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>pioglitazone</th><td>0</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>PF-00539745-00</th><td>0</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>valdecoxib</th><td>0</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>PF-00539758-00</th><td>0</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>orlistat</th><td>0</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>5707885</th><td>0</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>C-75</th><td>0</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>5155877</th><td>0</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>dinoprostone</th><td>0</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>5194442</th><td>0</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>AG-028671</th><td>0</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>6-bromoindirubin-3'-oxime</th><td>0</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>Gly-His-Lys</th><td>0</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>STOCK1N-35696</th><td>0</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>STOCK1N-28457</th><td>0</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>H-89</th><td>0</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>SB-202190</th><td>0</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>NS-398</th><td>0</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>SB-203580</th><td>0</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>AH-6809</th><td>0</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>alsterpaullone</th><td>0</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>SC-19220</th><td>0</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>0198306-0000</th><td>0</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>AR-A014418</th><td>0</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>semustine</th><td>0</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>CP-944629</th><td>0</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>irinotecan</th><td>0</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>BCB000039</th><td>0</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>11-deoxy-16,16-dimethylprostaglandin E2</th><td>0</td><td>1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 1288 × 2 of type dbl\n",
       "\\begin{tabular}{r|ll}\n",
       "  & Connectivity & P\\_value\\\\\n",
       "\\hline\n",
       "\tvalproic acid & -0.376625 & 0.003243431\\\\\n",
       "\trifabutin & -0.340380 & 0.003249652\\\\\n",
       "\tlevomepromazine & -0.329905 & 0.003363063\\\\\n",
       "\tpromazine & -0.338460 & 0.003408985\\\\\n",
       "\tPrestwick-559 & -0.332285 & 0.003453743\\\\\n",
       "\tmaprotiline & -0.345995 & 0.003483930\\\\\n",
       "\tnicergoline & -0.297975 & 0.003540995\\\\\n",
       "\tGW-8510 & -0.365775 & 0.003643482\\\\\n",
       "\tlasalocid & -0.287965 & 0.003653011\\\\\n",
       "\tpyrvinium & -0.312920 & 0.003692544\\\\\n",
       "\tnortriptyline & -0.346990 & 0.003785772\\\\\n",
       "\tprochlorperazine & -0.360245 & 0.003809080\\\\\n",
       "\tnorcyclobenzaprine & -0.319960 & 0.003935323\\\\\n",
       "\tfluphenazine & -0.351740 & 0.004089757\\\\\n",
       "\tLY-294002 & -0.358135 & 0.004118938\\\\\n",
       "\traloxifene & -0.301795 & 0.004218595\\\\\n",
       "\t4,5-dianilinophthalimide & -0.809330 & 0.006282675\\\\\n",
       "\ttrichostatin A & -0.330245 & 0.006335983\\\\\n",
       "\teucatropine & -0.288555 & 0.006421429\\\\\n",
       "\tsyrosingopine & -0.318895 & 0.006459413\\\\\n",
       "\tmebendazole & -0.300325 & 0.006618701\\\\\n",
       "\tlycorine &  0.319895 & 0.006644780\\\\\n",
       "\trescinnamine & -0.323295 & 0.006987249\\\\\n",
       "\treserpine & -0.285080 & 0.007184183\\\\\n",
       "\tsirolimus & -0.361685 & 0.007393165\\\\\n",
       "\ttrifluoperazine & -0.375195 & 0.007405285\\\\\n",
       "\tvorinostat & -0.378310 & 0.008683363\\\\\n",
       "\tmethotrexate & -0.336405 & 0.008864197\\\\\n",
       "\tfulvestrant & -0.281245 & 0.009065718\\\\\n",
       "\tphenazopyridine & -0.321425 & 0.009136916\\\\\n",
       "\t... & ... & ...\\\\\n",
       "\tPF-00562151-00 & 0 & 1\\\\\n",
       "\tpioglitazone & 0 & 1\\\\\n",
       "\tPF-00539745-00 & 0 & 1\\\\\n",
       "\tvaldecoxib & 0 & 1\\\\\n",
       "\tPF-00539758-00 & 0 & 1\\\\\n",
       "\torlistat & 0 & 1\\\\\n",
       "\t5707885 & 0 & 1\\\\\n",
       "\tC-75 & 0 & 1\\\\\n",
       "\t5155877 & 0 & 1\\\\\n",
       "\tdinoprostone & 0 & 1\\\\\n",
       "\t5194442 & 0 & 1\\\\\n",
       "\tAG-028671 & 0 & 1\\\\\n",
       "\t6-bromoindirubin-3'-oxime & 0 & 1\\\\\n",
       "\tGly-His-Lys & 0 & 1\\\\\n",
       "\tSTOCK1N-35696 & 0 & 1\\\\\n",
       "\tSTOCK1N-28457 & 0 & 1\\\\\n",
       "\tH-89 & 0 & 1\\\\\n",
       "\tSB-202190 & 0 & 1\\\\\n",
       "\tNS-398 & 0 & 1\\\\\n",
       "\tSB-203580 & 0 & 1\\\\\n",
       "\tAH-6809 & 0 & 1\\\\\n",
       "\talsterpaullone & 0 & 1\\\\\n",
       "\tSC-19220 & 0 & 1\\\\\n",
       "\t0198306-0000 & 0 & 1\\\\\n",
       "\tAR-A014418 & 0 & 1\\\\\n",
       "\tsemustine & 0 & 1\\\\\n",
       "\tCP-944629 & 0 & 1\\\\\n",
       "\tirinotecan & 0 & 1\\\\\n",
       "\tBCB000039 & 0 & 1\\\\\n",
       "\t11-deoxy-16,16-dimethylprostaglandin E2 & 0 & 1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 1288 × 2 of type dbl\n",
       "\n",
       "| <!--/--> | Connectivity | P_value |\n",
       "|---|---|---|\n",
       "| valproic acid | -0.376625 | 0.003243431 |\n",
       "| rifabutin | -0.340380 | 0.003249652 |\n",
       "| levomepromazine | -0.329905 | 0.003363063 |\n",
       "| promazine | -0.338460 | 0.003408985 |\n",
       "| Prestwick-559 | -0.332285 | 0.003453743 |\n",
       "| maprotiline | -0.345995 | 0.003483930 |\n",
       "| nicergoline | -0.297975 | 0.003540995 |\n",
       "| GW-8510 | -0.365775 | 0.003643482 |\n",
       "| lasalocid | -0.287965 | 0.003653011 |\n",
       "| pyrvinium | -0.312920 | 0.003692544 |\n",
       "| nortriptyline | -0.346990 | 0.003785772 |\n",
       "| prochlorperazine | -0.360245 | 0.003809080 |\n",
       "| norcyclobenzaprine | -0.319960 | 0.003935323 |\n",
       "| fluphenazine | -0.351740 | 0.004089757 |\n",
       "| LY-294002 | -0.358135 | 0.004118938 |\n",
       "| raloxifene | -0.301795 | 0.004218595 |\n",
       "| 4,5-dianilinophthalimide | -0.809330 | 0.006282675 |\n",
       "| trichostatin A | -0.330245 | 0.006335983 |\n",
       "| eucatropine | -0.288555 | 0.006421429 |\n",
       "| syrosingopine | -0.318895 | 0.006459413 |\n",
       "| mebendazole | -0.300325 | 0.006618701 |\n",
       "| lycorine |  0.319895 | 0.006644780 |\n",
       "| rescinnamine | -0.323295 | 0.006987249 |\n",
       "| reserpine | -0.285080 | 0.007184183 |\n",
       "| sirolimus | -0.361685 | 0.007393165 |\n",
       "| trifluoperazine | -0.375195 | 0.007405285 |\n",
       "| vorinostat | -0.378310 | 0.008683363 |\n",
       "| methotrexate | -0.336405 | 0.008864197 |\n",
       "| fulvestrant | -0.281245 | 0.009065718 |\n",
       "| phenazopyridine | -0.321425 | 0.009136916 |\n",
       "| ... | ... | ... |\n",
       "| PF-00562151-00 | 0 | 1 |\n",
       "| pioglitazone | 0 | 1 |\n",
       "| PF-00539745-00 | 0 | 1 |\n",
       "| valdecoxib | 0 | 1 |\n",
       "| PF-00539758-00 | 0 | 1 |\n",
       "| orlistat | 0 | 1 |\n",
       "| 5707885 | 0 | 1 |\n",
       "| C-75 | 0 | 1 |\n",
       "| 5155877 | 0 | 1 |\n",
       "| dinoprostone | 0 | 1 |\n",
       "| 5194442 | 0 | 1 |\n",
       "| AG-028671 | 0 | 1 |\n",
       "| 6-bromoindirubin-3'-oxime | 0 | 1 |\n",
       "| Gly-His-Lys | 0 | 1 |\n",
       "| STOCK1N-35696 | 0 | 1 |\n",
       "| STOCK1N-28457 | 0 | 1 |\n",
       "| H-89 | 0 | 1 |\n",
       "| SB-202190 | 0 | 1 |\n",
       "| NS-398 | 0 | 1 |\n",
       "| SB-203580 | 0 | 1 |\n",
       "| AH-6809 | 0 | 1 |\n",
       "| alsterpaullone | 0 | 1 |\n",
       "| SC-19220 | 0 | 1 |\n",
       "| 0198306-0000 | 0 | 1 |\n",
       "| AR-A014418 | 0 | 1 |\n",
       "| semustine | 0 | 1 |\n",
       "| CP-944629 | 0 | 1 |\n",
       "| irinotecan | 0 | 1 |\n",
       "| BCB000039 | 0 | 1 |\n",
       "| 11-deoxy-16,16-dimethylprostaglandin E2 | 0 | 1 |\n",
       "\n"
      ],
      "text/plain": [
       "                                        Connectivity P_value    \n",
       "valproic acid                           -0.376625    0.003243431\n",
       "rifabutin                               -0.340380    0.003249652\n",
       "levomepromazine                         -0.329905    0.003363063\n",
       "promazine                               -0.338460    0.003408985\n",
       "Prestwick-559                           -0.332285    0.003453743\n",
       "maprotiline                             -0.345995    0.003483930\n",
       "nicergoline                             -0.297975    0.003540995\n",
       "GW-8510                                 -0.365775    0.003643482\n",
       "lasalocid                               -0.287965    0.003653011\n",
       "pyrvinium                               -0.312920    0.003692544\n",
       "nortriptyline                           -0.346990    0.003785772\n",
       "prochlorperazine                        -0.360245    0.003809080\n",
       "norcyclobenzaprine                      -0.319960    0.003935323\n",
       "fluphenazine                            -0.351740    0.004089757\n",
       "LY-294002                               -0.358135    0.004118938\n",
       "raloxifene                              -0.301795    0.004218595\n",
       "4,5-dianilinophthalimide                -0.809330    0.006282675\n",
       "trichostatin A                          -0.330245    0.006335983\n",
       "eucatropine                             -0.288555    0.006421429\n",
       "syrosingopine                           -0.318895    0.006459413\n",
       "mebendazole                             -0.300325    0.006618701\n",
       "lycorine                                 0.319895    0.006644780\n",
       "rescinnamine                            -0.323295    0.006987249\n",
       "reserpine                               -0.285080    0.007184183\n",
       "sirolimus                               -0.361685    0.007393165\n",
       "trifluoperazine                         -0.375195    0.007405285\n",
       "vorinostat                              -0.378310    0.008683363\n",
       "methotrexate                            -0.336405    0.008864197\n",
       "fulvestrant                             -0.281245    0.009065718\n",
       "phenazopyridine                         -0.321425    0.009136916\n",
       "...                                     ...          ...        \n",
       "PF-00562151-00                          0            1          \n",
       "pioglitazone                            0            1          \n",
       "PF-00539745-00                          0            1          \n",
       "valdecoxib                              0            1          \n",
       "PF-00539758-00                          0            1          \n",
       "orlistat                                0            1          \n",
       "5707885                                 0            1          \n",
       "C-75                                    0            1          \n",
       "5155877                                 0            1          \n",
       "dinoprostone                            0            1          \n",
       "5194442                                 0            1          \n",
       "AG-028671                               0            1          \n",
       "6-bromoindirubin-3'-oxime               0            1          \n",
       "Gly-His-Lys                             0            1          \n",
       "STOCK1N-35696                           0            1          \n",
       "STOCK1N-28457                           0            1          \n",
       "H-89                                    0            1          \n",
       "SB-202190                               0            1          \n",
       "NS-398                                  0            1          \n",
       "SB-203580                               0            1          \n",
       "AH-6809                                 0            1          \n",
       "alsterpaullone                          0            1          \n",
       "SC-19220                                0            1          \n",
       "0198306-0000                            0            1          \n",
       "AR-A014418                              0            1          \n",
       "semustine                               0            1          \n",
       "CP-944629                               0            1          \n",
       "irinotecan                              0            1          \n",
       "BCB000039                               0            1          \n",
       "11-deoxy-16,16-dimethylprostaglandin E2 0            1          "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "p_value_sars\n"
   ]
  },
  
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [],
   "source": [
    "write.csv(p_value_sars,\"sars_results.csv\", row.names = T)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Based on p value and connectivity score (negative score means that drug reverses the disease signature) I have decided to do \n",
    "docking on valproic acid. This drug is FDA-approved for treatment of epilepsy, bipolar disorder  and migraine headeche. \n",
    "Valproic acid is also tested in the treatment of AIDS and cancer.\n",
    "I also in my results get that sirolimus can be used against coronaviruses, and study by Zhou et al get the similar results.\n",
    " \n",
    "References:\n",
    "Yadi Zhou, Yuan Hou, Jiayu Shen, Yin Huang, William Martin, Feixiong Cheng.Network-based Drug Repurposing for Human Coronavirus\n",
    "doi: https://doi.org/10.1101/2020.02.03.20020263\n",
    "\n",
    "O. Trott, A. J. Olson,AutoDock Vina: improving the speed and accuracy of dockingwith a new scoring function,\n",
    "efficient optimization and multithreading, Journal of Computational Chemistry 31 (2010); 455-461; DOI 10.1002/jcc.21334  \n",
    "\n",
    "Study of Valproic Acid to Treat HIV Infected Adults https://clinicaltrials.gov/ct2/show/NCT00312546\n",
    "\n",
    "Vázquez-Calvo A, Saiz JC, Sobrino F, Martín-Acebes MA. Inhibition of enveloped virus infection of cultured cells by valproic acid.\n",
    "J Virol. 2011;85(3):1267–1274. doi:10.1128/JVI.01717-10"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.6.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
